### PR TITLE
classfields: Tests for modified algorithms and Private Names

### PR DESCRIPTION
**Pull request by spectranaut**

### CODE REVIEW

File: test/language/statements/class/fielddefinition-initializer-abrupt-completion.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    2. Assert: fieldRecord is a Record as created by ClassFieldDefinitionEvaluation.
+    3. Let fieldName be fieldRecord.[[Name]].
+    4. Let initializer be fieldRecord.[[Initializer]].
+    5. If initializer is not empty, then
+        a.Let initValue be ? Call(initializer, receiver).
+
+features: [class-fields]
+---*/
+
+var f() {
```
Comment (bakkot) [inline test/language/statements/class/fielddefinition-initializer-abrupt-completion.js]: `function f`, possibly? Here and elsewhere.

Comment (spectranaut) [reply test/language/statements/class/fielddefinition-initializer-abrupt-completion.js]: whoops nice catch :)

File: test/language/statements/class/privatefieldget-success-1.js [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+features: [class-fields]
+---*/
+
+class Outer {
+  #x = 42;
+
+  innerclass() {
+    var self = this;
+
+    return class extend Outer {
```
Comment (bakkot) [inline test/language/statements/class/privatefieldget-success-1.js]: s/extend/extends/, here and in a few other places

Comment (spectranaut) [reply test/language/statements/class/privatefieldget-success-1.js]: thanks!

File: test/language/statements/class/privatefieldget-typeerror-4.js [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+    1. Assert: P is a Private Name value.
+    2. Assert: O is an object with a [[PrivateFieldValues]] internal slot.
+    3. For each element entry in O.[[PrivateFieldValues]],
+      a. If entry.[[PrivateName]] is P, return entry.
+    4. Return empty.
+
+features: [class-fields]
+---*/
+
+classfactory() {
```
Comment (bakkot) [inline test/language/statements/class/privatefieldget-typeerror-4.js]: `function classfactory() {`

File: test/language/statements/class/fielddefinition-initializer-abrupt-completion.js [hunk 4]
```diff
@@ -0,0 +1,46 @@
+// Copyright (C) 2017 Valerie Young. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+description: Class construction should error if evaluation of field initializer errors
+esid: [[construct]]
```
Comment (leobalter) [inline test/language/statements/class/fielddefinition-initializer-abrupt-completion.js]: this is wrong :)

File: test/language/statements/class/privatefieldadd-typeerror.js [hunk 5]
```diff
@@ -0,0 +1,60 @@
+// Copyright (C) 2017 Valerie Young. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+description: Class construction should error if attempting to add private field twice
+esid: [[construct]]
```
Comment (leobalter) [inline test/language/statements/class/privatefieldadd-typeerror.js]: same

File: test/language/statements/class/privatefieldget-success-1.js [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+
+class Outer {
+  #x = 42;
+
+  innerclass() {
+    var self = this;
+
+    return class extends Outer {
+      f() {
+        return self.#x;
```
Comment (leobalter) [inline test/language/statements/class/privatefieldget-success-1.js]: amazing

File: test/language/statements/class/privatefieldget-success-5.js [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+  f() {
+    return this.#x;
+  }
+}
+
+var c1 = new C();
+var c2 = new C();
+var value = c2.f.call(c1);
+
+assert.sameValue(value, 'Avalue')
```
Comment (leobalter) [inline test/language/statements/class/privatefieldget-success-5.js]: the final assertions are missing a semicolon at the end, but this should not be a blocker.

**Comment by spectranaut** [comment]
@leo pused fixes!

**Comment by gsathya** [comment]
Awesome work @spectranaut :100: I'm working on implementing private fields in V8 and these tests are spot on! Super helpful to have them :+1:

**Pull request merged**